app/core/Snaps/SnapsMethodMiddleware.ts

```diff
@@ -36,6 +37,11 @@ const snapMethodMiddlewareBuilder = (
       engineContext.PermissionController,
       origin,
     ),
+    hasPermission: engineContext.PermissionController.hasPermission(origin),
```

Comment:
```suggestion
    hasPermission: engineContext.PermissionController.hasPermission.bind(engineContext.PermissionController, origin),
```

This should fix the error you are seeing and bring this closer to the extension implementation!